enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,438 @@
+        implement the new profile?
+      - How would we aid developers implementing the new profile?
+### Drawbacks
+
+- Extra CI cycles
+
+## Design Details
+
+### Open Questions
+
+### Test Plan
+
+- Unit tests in CMO to validate that the correct monitors are being selected
+- E2E tests in CMO to validate that everything works correctly 
+- For the `minimal` profile, origin/CI test to validate that every metric used
+in a resource (Alerts/PrometheusRules/Dashboards) exist in the `keep` expression
+of a minimal monitors.
+
+### Graduation Criteria
+
+- Released as TechPreview: the default being `full`, it
+shouldn't impact operations.
+
+- Plan to GA: this would officially support the "minimal"
+profile out-of-the-box and removes the earlier-imposed
+TechPreview gate. PTAL at the section below for more details.
+
+#### Tech Preview -> GA
+
+- [Automation to update metrics in collection profiles](https://issues.redhat.com/browse/MON-3106)
+- [Telemetry signal for collection profile usage](https://issues.redhat.com/browse/MON-3231)
+- [Enhancement proposal for metrics collection profile](https://issues.redhat.com/browse/MON-2692)
+- [CLI tool to facilitate implementation of metrics collection profiles for OCP components](https://issues.redhat.com/browse/MON-2694)
+- [Drop TechPreview gate on collection profiles](https://issues.redhat.com/browse/MON-3215)
+- [origin/CI tool to validate collection profiles](https://issues.redhat.com/browse/MON-3105)
+- [User facing documentation created in openshift-docs](https://issues.redhat.com/browse/OBSDOCS-330)
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+- If metrics collection profiles is accepted and is released to 4.13 then we
+  must backport the new monitors selectors to 4.12. The reason being that when
+  an upgrade occurs and the new monitors are deployed, we will not want the 4.12
+  Prometheus-Operator to process all the new monitors as this would configure
+  the Prometheus instances to start double scraping the OpenShift components
+  that implemented the metrics collection profiles feature. Done in
+  [cluster-monitoring-operator#2047](https://github.com/openshift/cluster-monitoring-operator/pull/2047)
+- Once we backport the new monitors selectors upgrades and
+  downgrades are not expected to present a significant challenge.
+
+### Version Skew Strategy
+
+TBD but I don't think it applies here
+
+### minimal Aspects of API Extensions
+
+TBD but I don't think it applies here
+
+#### Failure Modes
+
+TBD but I don't think it applies here
+
+#### Support Procedures
+
+TBD but I don't think it applies here
+
+## Implementation History
+
+Initial proofs-of-concept:
+
+- https://github.com/openshift/cluster-monitoring-operator/pull/1785
+
+## Alternatives
+
+- Make CMO injecting metric relabelling for all service monitors based on the
```

Comment:
I'm not even sure how it can work in practice.

Reply:
I had put this here just to show the reviewers that it had been something we had considered. But if you feel it doesn't make sense to have here I'm fine with removing it

Reply:
Right, I'm ok keeping it here but maybe highlight how cumbersome and error-prone it would be. The only way I could see it working would be that CMO watches the component monitors and generates new resources that would be reconciled by prometheus-operator instead of the original ones.